woocommerce/woocommerce
# Autoload test helpers and exclude from test suites.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





The change takes two helper classes (well, one is actually a trait):

- `HposTestCase`
- `HPOSToggleTrait`

And places them in the same directory. It also excludes that directory from the test suite, and adds support for loading them via the bootstrap class autoloader. Reasons for this:

- Avoids us manually requiring those helper classes.
- Eliminates a warning about `HposTestCase` not containing any tests. While harmless, this was also unnecessary noise.
- Finally, though not a problem I saw anyone else reporting (and not a problem in CI so far as I can tell), it solves something I was encountering locally in which the `DataSynchronizerTests` (which inherits `HposTestCase`) was loaded _before_ `HposTestCase`, resulting in a class-not-found error.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Given the nature of the change, code review is sufficient. Of course, the test suite should also pass (and we're actively working on some unrelated problems that are causing a different set of failures). Keeping as a draft initially.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment Improves organization of test helper code; does not merit its own changelog entry.

</details>
